File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/template_typedef_runme.m
template_typedef

d = make_Identity_float();
c = make_Identity_reald();


try
  a = swig_this(d);
  a = swig_this(c);
catch
  error
end_try_catch

try
  e = make_Multiplies_float_float_float_float(d, d);
  a = swig_this(e);
catch
  e
  error("is not an instance")
end_try_catch

try
  f = make_Multiplies_reald_reald_reald_reald(c, c);
  a = swig_this(f);
catch
  f
  error("is not an instance");
end_try_catch

try
  g = make_Multiplies_float_float_reald_reald(d, c);
  a = swig_this(g);
catch
  error(g, "is not an instance")
end_try_catch


# the old large format
if (strcmp("<unknown>",swig_typequery("vfncs::ArithUnaryFunction<vfncs::arith_traits<float,double>::argument_type,vfncs::arith_traits<float,double >::result_type > *")))
  error
endif

# the reduced format
if (strcmp("<unknown>",swig_typequery("vfncs::ArithUnaryFunction<double,double> *")))
  error
endif

# this is a bad name
if (!strcmp("<unknown>",swig_typequery("vfncs::ArithUnaryFunction<double,doublex> *")))
  error
endif

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/template_typedef_runme.py
from template_typedef import *

d = make_Identity_float()
c = make_Identity_reald()


try:
    a = d.this
    a = c.this
except:
    raise RuntimeError

try:
    e = make_Multiplies_float_float_float_float(d, d)
    a = e.this
except:
    print e, "is not an instance"
    raise RuntimeError

try:
    f = make_Multiplies_reald_reald_reald_reald(c, c)
    a = f.this
except:
    print f, "is not an instance"
    raise RuntimeError

try:
    g = make_Multiplies_float_float_reald_reald(d, c)
    a = g.this
except:
    print g, "is not an instance"
    raise RuntimeError


# the old large format
if not SWIG_TypeQuery("vfncs::ArithUnaryFunction<vfncs::arith_traits<float,double>::argument_type,vfncs::arith_traits<float,double >::result_type > *"):
    raise RuntimeError


# the reduced format
if not SWIG_TypeQuery("vfncs::ArithUnaryFunction<double,double> *"):
    raise RuntimeError

# this is a bad name
if SWIG_TypeQuery("vfncs::ArithUnaryFunction<double,doublex> *"):
    raise RuntimeError
